packages/react-instantsearch-core/src/connectors/connectMenu.js

```diff
@@ -135,18 +134,24 @@ export default createConnector({
       };
     }
 
-    const items = isFromSearch
-      ? searchForFacetValuesResults[attribute].map(v => ({
-          label: v.value,
-          value: getValue(v.value, props, searchState, {
-            ais: props.contextValue,
-            multiIndexContext: props.indexContextValue,
-          }),
-          _highlightResult: { label: { value: v.highlighted } },
-          count: v.count,
-          isRefined: v.isRefined,
-        }))
-      : results.getFacetValues(attribute, { sortBy }).map(v => ({
+    let items;
+    if (isFromSearch) {
+      items = searchForFacetValuesResults[attribute].map(v => ({
+        label: v.value,
+        value: getValue(v.value, props, searchState, {
+          ais: props.contextValue,
+          multiIndexContext: props.indexContextValue,
+        }),
+        _highlightResult: { label: { value: v.highlighted } },
+        count: v.count,
+        isRefined: v.isRefined,
+      }));
+    } else {
+      items = results
+        .getFacetValues(attribute, {
+          sortBy: searchable ? undefined : defaultSortBy,
```

Comment:
Rather than relying on the default one from the Helper (which might change). We could specify the sort that we want to apply in this case.

Reply:
we will keep this as undefined, because the other widgets also pass undefined